### ISSUE TITLE
Support protected/package level classes.

### ISSUE THREAD

**Issue by autovalue** [yshrsmz/simple-preferences#10]
On [this line](https://github.com/yshrsmz/simple-preferences/blob/a7b7ac0d1befb0d17357d6b0a723d0fadc3c29f4/processor/src/main/java/net/yslibrary/simplepreferences/processor/SimplePreferencesProcessor.java#L118) you're doing a check for public visibility, whereas it'd make more sense to check if it's flagged as a private class.

My config file is as follows:
```
@Preferences(value = Config.FILE_NAME)
abstract class Config {
    static final String FILE_NAME = "my_config_pref";

    @Key
    int userId = 0;

    @Key
    String userName = "";

    @Key(name = "is_premium", omitGetterPrefix = true)
    boolean isPremium = false;
}
```

Given the generated `ConfigPrefs` class is in the same package, package level extends should be supported preventing me from exposing both `Config` and `ConfigPrefs` publicly.

**Comment by yshrsmz** [comment]
Hi, thanks for reporting.
That makes sense. 
I'm planning to just inherit parent visibility.

**Comment by autovalue** [comment]
Inheriting the parent's visibility will force the user to publicly expose both `Config` `ConfigPrefs`, or hide both at the package level (which isn't what we're after).

I think the `ConfigPref` should be exposed publicly by default.

**Comment by yshrsmz** [comment]
@autovalue 
makes sense.
will look into this this week.

Since it seems that top level classes can only be public or package private, what about adding `expose`(field name may change) option to `@Preferences`, to control visibility of generated classes?

```java
@Preferences(value = Config.FILE_NAME,
             expose = false)
abstract class Config {
    static final String FILE_NAME = "my_config_pref";

    @Key
    int userId = 0;

    @Key
    String userName = "";

    @Key(name = "is_premium", omitGetterPrefix = true)
    boolean isPremium = false;
}

// `expose = false` will generate package private ConfigPrefs
class ConfigPrefs extends Conifg {
  // getter/setter ...
}
```

**Comment by autovalue** [comment]
@yshrsmz Yes, that would be perfect! thanks

**Comment by yshrsmz** [comment]
@autovalue released as 1.4.0

**Comment by autovalue** [comment]
Thanks.
